exercises/practice/spiral-matrix/test_spiral_matrix.c

```diff
@@ -0,0 +1,120 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include "test-framework/unity.h"
+#include "spiral_matrix.h"
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static bool matrix_matches(int expected_size,
+                           int expected_matrix[][expected_size],
+                           const spiral_matrix_t *actual)
+{
+   if (expected_size != actual->size)
+      return false;
+
+   for (int i = 0; i < expected_size; i++) {
+      for (int j = 0; j < expected_size; j++) {
+         if (expected_matrix[i][j] != actual->matrix[i][j]) {
+            printf("expected[%d][%d] = %d\n", i, j, expected_matrix[i][j]);
+            printf("actual[%d][%d] = %d\n", i, j, actual->matrix[i][j]);
```

Comment:
Because the check is pretty opaque (it tells you whether it matches, but that's it), I added in these prints to show where the first difference is when there's a failure.

Reply:
How does this look when the tests run and have their standard output?
Would it interfere with how the web IDE detects pass/fail ?

Reply:
With this, the output for a failure is:

```
expected[0][0] = 1
actual[0][0] = 0
./test_spiral_matrix.c:52:test_trivial_spiral:FAIL: Expected TRUE Was FALSE
```

Without it, the output is:

```
./test_spiral_matrix.c:52:test_trivial_spiral:FAIL: Expected TRUE Was FALSE
```

>  Would it interfere with how the web IDE detects pass/fail ?

I don't think it will since the failure is still present, but I'm not sure whether it will show the console output in addition to the failure text.

Looking through the Unity docs I think I should be able to compare each row in the array. This should give enough context to understand where the failure is because each row is unique.

Reply:
With Unity's array matcher, the output is:

```
./test_spiral_matrix.c:22:test_spiral_of_size_5:FAIL: Element 2 Expected 25 Was 0
```

I think that's sufficient to work out where the error is even without knowing the row number.